.github/workflows/ci-docker.yml

```diff
@@ -0,0 +1,45 @@
+name: Build & Publish Docker Images
+
+on:
+  workflow_call:
+    secrets:
+      DOCKERHUB_USERNAME:
+        required: true
+      DOCKERHUB_TOKEN:
+        required: true
+
+# runs in ci-master after successful checks
+# you can use images built by this action in future jobs.
+# https://docs.docker.com/build/ci/github-actions/examples/#share-built-image-between-jobs
+jobs:
+  docker:
+    # https://github.com/marketplace/actions/build-and-push-docker-images
+    runs-on: ubuntu-latest
+    steps:
+      # ensure working with latest code
+      - name: Checkout
+        uses: actions/checkout@v3
+      # generate a git commit hash to be used as image tag
+      - name: Generate short hash
+        id: commit-hash
+        run: echo "short=$( git rev-parse --short $GITHUB_SHA )" >> $GITHUB_OUTPUT
+      # qemu is used to emulate different platform architectures
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      # cross-platform build of the image
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      # authenticate for publish to docker hub
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
added this as `CI_DOCKERHUB_TOKEN` to namespace it for future proofing if other dockerhub actions and secrets are added